client/src/app/components/simple-document-viewer/SimpleDocumentViewer.tsx

```diff
@@ -22,63 +26,133 @@ type ControlledEditor = {
-   * the download file action.  Defaults to `Language.yaml`.
-   */
-  language?: Language.yaml | Language.json;
-
   /**
    * Height of the document viewer, or `"full"` to take up all of the available
    * vertical space.  Defaults to "450px".
    */
   height?: string | "full";
+
+  /** Callback triggered when user selects a new document to display */
+  onDocumentChange?: (documentId: DocumentId) => void;
 }
 
+export type DocumentId = number | "LOG_VIEW" | "MERGED_VIEW";
+
+const useDocuments = ({
+  taskId,
+  selectedId,
+  currentLanguage,
+}: {
+  taskId?: number;
+  selectedId: DocumentId;
+  currentLanguage: Language;
+}) => {
+  const { task, refetch: refetchTask } = useFetchTaskByIdAndFormat({
+    taskId,
+    format: currentLanguage === Language.yaml ? "yaml" : "json",
+    enabled:
+      !!taskId && (selectedId === "LOG_VIEW" || selectedId === "MERGED_VIEW"),
+    merged: selectedId === "MERGED_VIEW",
+  });
+
+  const isAttachment = typeof selectedId === "number";
+  const { attachment, refetch: refetchAttachment } = useFetchTaskAttachmentById(
+    {
+      attachmentId: isAttachment ? selectedId : undefined,
+      enabled: isAttachment,
+    }
+  );
+
+  return isAttachment
+    ? { code: attachment, refetch: refetchAttachment }
+    : { code: task, refetch: refetchTask };
+};
+
 /**
  * Fetch and then use the `@patternfly/react-code-editor` to display a document in
  * read-only mode with language highlighting applied.
  */
 export const SimpleDocumentViewer = ({
-  documentId,
+  taskId,
+  attachmentId,
+  attachments,
   downloadFilename,
-  language = Language.yaml,
   height = "450px",
+  onDocumentChange,
 }: ISimpleDocumentViewerProps) => {
-  const editorRef = React.useRef<ControlledEditor>();
-  const [currentLanguage, setCurrentLanguage] = React.useState(language);
-  const [code, setCode] = React.useState<string>();
-  const [merged, setMerged] = React.useState(false);
-
-  const { task, isFetching, fetchError, refetch } = useFetchTaskByID(
-    documentId,
-    currentLanguage === Language.yaml ? "yaml" : "json",
-    merged
+  const configuredDocuments: Document[] = [
+    {
+      id: "LOG_VIEW",
+      name: "Log view",
+      isSelected: !attachmentId,
+      languages: [Language.yaml, Language.json],
+    },
+    {
+      id: "MERGED_VIEW",
+      name: "Merged log view",
+      description: "with inlined commands output",
+      isSelected: false,
+      languages: [Language.yaml, Language.json],
+    },
+    ...attachments.map(({ id, name }) => ({
+      id,
+      name,
+      isSelected: id === attachmentId,
+      languages: [
+        Language.plaintext,
+        name.endsWith(".yaml") && Language.yaml,
+        name.endsWith(".json") && Language.json,
+      ].filter(Boolean),
+    })),
+  ];
+
+  const [documents, setDocuments] = React.useState([...configuredDocuments]);
+  const selectedId =
+    documents.find(({ isSelected }) => isSelected)?.id ?? "LOG_VIEW";
```

Comment:
What about just using `selectedId` as the state instead of keeping the documents in state?  You can memoize `documents` based on `attachments` and `selectedId` and keep things a bit simpler.  Simpler by just computing the document attributes in one place.

Else, the initial state can just be `React.useState(configuredDocuments)`.  No need to make a copy of the array like that.

Reply:
I've switched to `selectId` as you suggested - some things look simpler but at cost of extra checks if selected ID is "in range" (has a matching document).